api/src/main/java/marquez/common/Utils.java

```diff
@@ -112,7 +117,46 @@ public static UUID toUuid(@NonNull final String uuidString) {
     return UUID.fromString(uuidString);
   }
 
-  public static Instant toInstant(@Nullable final String asIso) {
+  public static Instant toInstantOrNull(@Nullable final String asIso) {
     return (asIso == null) ? null : Instant.from(ISO_INSTANT.parse(asIso));
   }
+
+  public static SourceType sourceTypeFor(@NonNull final URI connectionUrl) {
+    return SourceType.of(toUrlWithNoJdbcPrefix(connectionUrl).getScheme());
+  }
+
+  public static URI urlWithNoCredentials(@NonNull final URI connectionUrl) {
+    final URI urlWithNoJdbcPrefix = toUrlWithNoJdbcPrefix(connectionUrl);
+    try {
+      return new URI(
+          (isJdbcUrl(connectionUrl))
+              ? JDBC + urlWithNoJdbcPrefix.getScheme()
+              : urlWithNoJdbcPrefix.getScheme(),
+          null,
+          urlWithNoJdbcPrefix.getHost(),
+          urlWithNoJdbcPrefix.getPort(),
+          urlWithNoJdbcPrefix.getPath(),
+          urlWithNoJdbcPrefix.getQuery(),
+          urlWithNoJdbcPrefix.getFragment());
+    } catch (URISyntaxException e) {
+      final AssertionError error = new AssertionError("Malformed URI: " + connectionUrl);
+      error.initCause(e);
+      throw error;
+    }
+  }
+
+  static boolean isJdbcUrl(@NonNull final URI connectionUrl) {
+    final String connectionUrlString = connectionUrl.toString();
+    return connectionUrlString.startsWith(JDBC);
+  }
+
+  static URI toUrlWithNoJdbcPrefix(@NonNull final URI connectionUrl) {
+    return isJdbcUrl(connectionUrl)
+        ? URI.create(dropJdbcPrefix(connectionUrl.toString()))
+        : connectionUrl;
+  }
```

Comment:
This would fail with `jdbc:microsoft:sqlserver://` as a prefix. This is also an issue with the openlineage code location code so we may benefit from our own parser class.